src/prettyprint.lua

```diff
@@ -0,0 +1,229 @@
+local COLOURS = require("src.utils.consoleColor")
+local PASS_SYMBOL = COLOURS.PASS("√")
+local ERROR_SYMBOL = "●"
+local FAIL_SYMBOL = COLOURS.FAIL("×")
+local PASS_HEADER = COLOURS.PASS_HEADER(" PASS ")
+local FAIL_HEADER = COLOURS.FAIL_HEADER(" FAIL ")
+
+-- Matches Jest
+local TAB_SIZE = 2
+
+local NodeType = require("src.interface.testnodetype")
+local tablex = require("src.utils.tablex")
+local prettyValue = require("src.utils.prettyValue")
+
+local function getTabs(amount)
+	return (" "):rep(amount * TAB_SIZE)
+end
+
+--- Recursively traverses the node tree while calling the supplied callback.
+---@param node lest.TestResult|lest.Describe
+---@param onNode fun(node: lest.TestResult|lest.Describe, depth: number):boolean A callback which is called for each Describe and Test in the tree. You can return true to keep going or return false to terminate.
+---@param onEnd fun()|nil A callback which is called once a branch in the tree has finished being traversed.
+local function traverseNodes(node, onNode, onEnd, _depth)
+	_depth = _depth or 0
+
+	for _, childNode in ipairs(node) do
+		-- If the callback does not want to traverse this specific branch anymore, it can request it by returning false.
+		if
+			onNode(childNode, _depth)
+			and childNode.type == NodeType.Describe
+		then
+			traverseNodes(childNode, onNode, onEnd, _depth + 1)
+		end
+	end
+
+	if onEnd then
+		onEnd()
+	end
+end
+
+--- Prints a simple summary about the tests
+---@param results lest.TestSuiteResults
+local function printSummary(results)
+	local passedTests, failedTests, totalTests = 0, 0, 0
+	local passedSuites, failedSuites, totalSuites = 0, 0, #results
+
+	for _, testSuite in ipairs(results) do
+		if testSuite.pass then
+			passedSuites = passedSuites + 1
+		else
+			failedSuites = failedSuites + 1
+		end
+	end
+
+	for _, testSuite in ipairs(results) do
+		traverseNodes(testSuite, function(node, _)
+			if node.type == NodeType.Test then
+				totalTests = totalTests + 1
+				if node.pass then
+					passedTests = passedTests + 1
+				else
+					failedTests = failedTests + 1
+				end
+			end
+
+			return true
+		end)
+	end
```

Comment:
Why are you looping over the same array twice.

Also there's no need to increment both `passedTests` and `failedTests`, you only need one and can calculate the other with `total*` and basic arithmetic.